server_src/routes/printers.js

```diff
@@ -131,7 +131,8 @@ router.post("/printerInfo", ensureAuthenticated, async (req, res) => {
     res.send(printers);
   } else {
     const index = _.findIndex(printers, function (o) {
-      return o._id === id;
+      //Make sure ID's are both strings to stop recursion issues
+      return JSON.stringify(o._id) === JSON.stringify(id);
```

Comment:
Stringifying objects... ok ehm wow sure. It is indeed a fix. Object.Is(obj1, obj2) is as well.

Reply:
I could get rid of the === and put == but I feel that was a worse solution.

Reply:
I feel that can be looked at at a later date, currently file manager isn't working which is a biggy. Although I do agree, but I haven't had chance to update my old habbits yet.

Reply:
Ok as a promise from my V2 side: I will fix it on the API and ensure everything is string, ok? Then Im ok with merging.

Reply:
Well the way I see it currently is that call is from farmPrinters and that's in state. That's a mess XD, I'm relying here on the fact that that always get's dumped with an ID which in this situation it's going to be because if it hasn't something else will have failed before this always get's called. 

I wouldn't see if as an express issue personally with how the code is laid out but happy to be proved wrong. It needs pulling out of the database as a string permanently but I feel there's somewhere in the code where it's not getting done and finding that will be difficult. 

I to me have to options in this situation and that's make sure it's a string here as a "patch" only until we rip apart the state file and the rest, as you'll know when you update and object that's copied it also updates the original. Or let the double "==" handle it and call it a day?